### ISSUE TITLE
redundant_clone false positive

### ISSUE THREAD

**Issue by ehuss** [rust-lang/rust-clippy#4633]
Clippy version: clippy 0.0.212 (648e5b90 2019-10-01)

I'm getting a false-positive for `redundant_clone` on [this line](https://github.com/rust-lang/cargo/blob/a429e8cc4614a46a86322a0777a477e2baa83f1c/src/cargo/core/package_id.rs#L159).  Here is a semi-reduced example:

```rust
#![allow(unused)]
#![warn(clippy::redundant_clone)]

#[derive(Clone)]
struct Version;

struct PackageId {
    inner: &'static PackageIdInner,
}

struct PackageIdInner {
    name: &'static str,
    version: Version,
}

impl PackageId {
    fn foo(name: &'static str, version: Version) {}

    fn bar(self) {
        PackageId::foo(self.inner.name, self.inner.version.clone());
        //                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^ redundant clone
        // Removing clone causes:
        // ERR: cannot move out of `self.inner.version` which is behind a shared reference
    }
}
```

Some more examples: 
- https://github.com/rust-lang/cargo/blob/a429e8cc4614a46a86322a0777a477e2baa83f1c/src/cargo/core/compiler/compilation.rs#L132
- https://github.com/rust-lang/cargo/blob/a429e8cc4614a46a86322a0777a477e2baa83f1c/src/cargo/util/toml/mod.rs#L1532-L1533

**Comment by flip1995** [comment]
#4509 fixed this, but didn't land in nightly yet. Should be fixed in next nightly.
